src/AstTypeAnalysis.cpp

```diff
@@ -740,21 +744,29 @@ void TypeAnalysis::run(const AstTranslationUnit& translationUnit) {
     auto* typeEnvAnalysis = translationUnit.getAnalysis<TypeEnvironmentAnalysis>();
     for (const AstRelation* rel : translationUnit.getProgram()->getRelations()) {
         for (const AstClause* clause : rel->getClauses()) {
-            // Perform the type analysis
-            std::map<const AstArgument*, TypeSet> clauseArgumentTypes = analyseTypes(
-                    typeEnvAnalysis->getTypeEnvironment(), *clause, translationUnit.getProgram());
-            argumentTypes.insert(clauseArgumentTypes.begin(), clauseArgumentTypes.end());
-
             if (!Global::config().get("debug-report").empty()) {
+                // Perform the type analysis and keep logs
```

Comment:
Could you instead make a variable for the debug output (nullptr/&analysisLogs) and call analyseTypes with that? That will simplify any future updates since only one method call will need updating.

Reply:
Fixed in e16185d10dee7789f65b1c444dce58994ab769ea, though if the result of `Global::config().get("debug-report").empty()` doesn't change during the loop this could be optimised.

Reply:
That value will never change at this point so it's safe to move it out of the loop - which also makes the logic clearer.